src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -690,7 +690,7 @@ public void updateResourceInStateIndex(
         Script script = new Script(
             ScriptType.INLINE,
             "painless",
-            "ctx._source.resources_created.add(params.newResource)",
+            "ctx._source.resources_created.add(params.newResource); if(ctx._source.containsKey('error')){ctx._source.remove('error')}",
```

Comment:
Great catch, I'll probably move this error removal when updating the state index prior to executing reprovision.

Reply:
Just to confirm if we have an error on model creation step which is a node that gets executed later, and we are only updating the index, don't we want the error to still be there or the error will get recreated because the model creation step will get executed again cause it failed in the first place

Reply:
If we have an error on a specific step, what would be the point of updating any other step there? In this case don't we want to deprovision the template completely and provision again?
Or let user know to deprovision and then provision again before reprovisioning?

Reply:
Any error will terminate the workflow. All the futures will get cancelled.  However, if a step is executing in parallel, it will go all the way until its future is completed (which will be a no-op after cancelling).

Reply:
I have modified the reprovision transport action to check error when retrieving state. Prior to execution of the reprovision sequence we remove the error field if it is there